k8s/setup.go

```diff
@@ -0,0 +1,82 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SetupConfig struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+type Setup struct {
```

Comment:
It's more of a `Client` IMO.

Reply:
`k8s.Setup` reads better than `k8s.Client`. The latter is misleading IMO.

Reply:
OK. Go ahead.